Samba - lib:tevent - add kqueue-based libtevent backend

anodos325 opened the pull request:
Initial work on kqueue-based libtevent backend for Samba. Based
on existing epoll backend, but accounting for differences between
epoll and kqueue APIs and behaviors. This PR just provides
bare minimum functionality for the backend. Future work will
add support for AIO kevents to libtevent and take advantage
of additional features unique to kqueue.

KQUEUE_IS_MPX_EVENT indicates that more than one tevent fd
event shares the same kevent.

KQUEUE_HAS_ADDITIONAL_KEVENT the tevent fd event has both
TEVENT_FD_READ and TEVENT_FD_WRITE flags, and therefore required
two separate kevents. This is used to indicate that we need
to check array of returned kevents to ensure that the correct flags
are sent to the fd event's handler function.

william-gr reviewed (approved):
I probably didnt review it as carefully as I should : P

anodos325 committed:
Samba - lib:tevent - add kqueue-based libtevent backend

Initial work on kqueue-based libtevent backend for Samba. Based
on existing epoll backend, but accounting for differences between
epoll and kqueue APIs and behaviors. This PR just provides
bare minimum functionality for the backend. Future work will
add support for AIO kevents to libtevent and take advantage
of additional features unique to kqueue.

KQUEUE_IS_MPX_EVENT indicates that more than one tevent fd
event shares the same kevent.

KQUEUE_HAS_ADDITIONAL_KEVENT the tevent fd event has both
TEVENT_FD_READ and TEVENT_FD_WRITE flags, and therefore required
two separate kevents. This is used to indicate that we need
to check array of returned kevents to ensure that the correct flags
are sent to the fd event's handler function.

anodos325 committed:
Simplify kqueue-related code. Remove loops.
